First pass at upgrading to QP

alexolivier opened the pull request:
- [x] Basic eCommerce Spec
- [x] Transaction Template
- [x] ~~Handle Variant Change~~ Can't be added via theme - details in docs
- [x] Update implementation docs

alexolivier committed:
First pass at upgrading to QP

alexolivier committed:
Create qubit-events.liquid

anthonydelage commented on qubit-events.liquid:
```diff
@@ -0,0 +1,200 @@
+        	value: {{ product.price | divided_by: 100 }}
+     	 },
+      	url: "{{ product.url }}",
+     	description: {{ product.description | strip_html | truncate: 100 | json }},
```
Why are we truncating at 100 characters? The QP `Text` type goes up to 2048 chars.

alexolivier replied:
Copied from old version

anthonydelage commented:
Do we have a sandbox we can test this in?

anthonydelage commented on qubit-events.liquid:
```diff
@@ -0,0 +1,200 @@
+     	 },
+      	url: "{{ product.url }}",
+     	description: {{ product.description | strip_html | truncate: 100 | json }},
+      	category: ["{{ product.type}}"],
```
Is there any way to convert this to the new field, `categories`?

anthonydelage commented on qubit-events.liquid:
```diff
@@ -0,0 +1,200 @@
+     	 },
+      	url: "{{ product.url }}",
+     	description: {{ product.description | strip_html | truncate: 100 | json }},
+      	category: ["{{ product.type}}"],
```
Is there any way to convert this to the new field, `categories`?

alexolivier replied:
The shopify data is a product type and a set of tags

alexolivier commented:
Added @adelage to our dev site uv-dev.myshopify.com

alexolivier committed:
Update qubit-events.liquid

alexolivier committed:
add transaction and update readme

anthonydelage commented on README.md:
```diff
@@ -1,16 +1,13 @@
-1. Add the Opentag container to the `<head>` of every page
-2. Add the Universal Variable (UV) directly before the Opentag container on every page.
+1. Add the Smartserve container to the `<head>` of every page
+2. Emit emits based on user browsing behaviour
```
Does the UV API need to be added in the head, or are we assuming that Smartserve will load before the events start firing?

alexolivier replied:
SS has aync on it so no guarantees so keep UV API

anthonydelage replied:
I'd add a blurb about including the UV API script then, as per [our docs](https://docs.qubit.com/implementing-qp-ec.html).

alexolivier commented:
SS has aync on it so no guarantees so keep UV API

On Fri, 16 Jun 2017, 18:51 Anthony Delage, <notifications@github.com> wrote:

> *@adelage* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/QubitProducts/shopify-universal-variable-asset/pull/8#discussion_r122497754>
> :
>
> >
> -1. Add the Opentag container to the `<head>` of every page
> -2. Add the Universal Variable (UV) directly before the Opentag container on every page.
> +1. Add the Smartserve container to the `<head>` of every page
> +2. Emit emits based on user browsing behaviour
>
> Does the UV API need to be added in the head, or are we assuming that
> Smartserve will load before the events start firing?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/QubitProducts/shopify-universal-variable-asset/pull/8#pullrequestreview-44623782>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABrdFpE6cUSs34-LNAlYkB9Kuz8lg3FQks5sEsCDgaJpZM4N7acW>
> .
>

alexolivier committed:
switch to categories

anthonydelage commented on qubit-events-transaction.liquid:
```diff
@@ -60,7 +60,7 @@
         description: {{ item.product.description | strip_html | truncate: 2048 | json }},
-        category: [{{ item.product.type | json}}],
+        categories: [{{ item.product.type | json}}],
         images: [{{ item.image | json }}]
```
@alexolivier is it actually `image` when it's an `item`, and `images` when it's a `product`?

alexolivier replied:
Looks like it in the Shopify API

alexolivier committed:
add more docs

alexolivier committed:
add readme for variant change

alexolivier committed:
more docs

alexolivier committed:
better copy

alexolivier committed:
Update README.md

anthonydelage commented on README.md:
```diff
@@ -1,63 +1,87 @@
+      stock: {{ product.inventory_quantity }},
+      price: {
+        currency: "{{ shop.currency }}",
+        value: {{ product.price | divided_by: 100 }}
```
Is the standard price format multiplied by 100? We might want to use a [money filter](https://help.shopify.com/themes/liquid/filters/money-filters) here instead.

anthonydelage commented on qubit-events-transaction.liquid:
```diff
@@ -0,0 +1,87 @@
+  }
+
+  var transactionSummary = {
+    id: "{{ order_number }}"
```
Any reason why we don't have the `checkout.` prefix here?

alexolivier committed:
Update qubit-events.liquid

Commit:
Updated readme

Added a reminder about updating the transaction snippet

anthonydelage committed:
Add minor updates

- Proper URL formatting for products and images
- Add more prices to transactions
- Emit ecBasketSummary events
- Update page types
